Automattic/woocommerce-payments
# Don't try load Subscriptions Core when WooCommerce isn't active

Fixes #3265 

#### Changes proposed in this Pull Request


- Don't init `woocommerce-subscriptions-core` if the `Automattic\WooCommerce\Admin\PluginsHelper` class doesn't exist.



When you deactivate the WooCommerce plugin and you have WooCommerce Payments active and the WCPay Subscriptions feature is enabled you get the following fatal error:

```
[02-Nov-2021 05:18:52 UTC] PHP Fatal error:  Uncaught Error: Class 'Automattic\WooCommerce\Admin\PluginsHelper' not found in /Users/matt/local/woo/wp-content/plugins/woocommerce-payments/woocommerce-payments.php:131
Stack trace:
#0 /Users/matt/local/woo/wp-content/plugins/woocommerce-payments/woocommerce-payments.php(134): {closure}('woocommerce-sub...')
#1 /Users/matt/local/woo/wp-content/plugins/woocommerce-payments/woocommerce-payments.php(152): wcpay_init_subscriptions_core()
#2 /Users/matt/local/woo/wp-settings.php(409): include_once('/Users/matt/loc...')
#3 /Users/matt/local/woo/wp-config.php(99): require_once('/Users/matt/loc...')
#4 /Users/matt/local/woo/wp-load.php(50): require_once('/Users/matt/loc...')
#5 /Users/matt/local/woo/wp-admin/admin.php(34): require_once('/Users/matt/loc...')
#6 /Users/matt/local/woo/wp-admin/plugins.php(10): require_once('/Users/matt/loc...')
#7 /Users/matt/.composer/vendor/laravel/valet/server.php(235): require('/Users/matt/loc...')
#8 {main}
  thrown in /Users/matt/local/woo/wp-content/plugins/woocommerce-payments/woocommerce-payments.php on line 131
```

This PR stops this fatal error from occurring by making sure the `Automattic\WooCommerce\Admin\PluginsHelper` class exists before trying to call `PluginsHelper::is_plugin_active()`. If this class doesn't exist, we don't try load Subscriptions Core.






#### Testing instructions





1. Activate WooCommerce and WooCommerce Payments
2. Remove `define( 'WCPAY_TEST_ENV', true );` from wp-config if you have this (this forces `WC_Payments::check_plugin_dependencies()` to always pass true
2. Make sure you have WCPay Subscriptions feature enabled (using WCPay dev tools or set `_wcpay_feature_subscriptions` option to `1`)
3. Go to **WP Admin > Plugins** and deactivate the WooCommerce plugin
4. On `trunk` you'll get fatal error
5. On this branch you won't see any fatal error and will see the following notice:

![image](https://user-images.githubusercontent.com/2275145/139793359-5992f900-8b60-4d28-a041-5ed0096d00fd.png)

---

Closes #3265
